OMCompiler/SimulationRuntime/c/util/utility.c

```diff
@@ -299,14 +299,14 @@ extern modelica_string OpenModelica_uriToFilename_impl(threadData_t *threadData,
     }
     if (resourcesDir) {
       if (strlen(dir)+2+strlen(resourcesDir) >= PATH_MAX) {
-        omc_assert_warning(info, "Path longer than PATH_MAX: %s/%s, ignoring the resourcesDir", dir, resourcesDir);
+        omc_assert_warning(info, "Path longer than PATH_MAX: %s/%s, ignoring the resourcesDir", MMC_STRINGDATA(dir), resourcesDir);
       } else {
-        int dirExists = 0==stat(dir, &stat_buf);
-        sprintf(buf, "%s/%s", dir, resourcesDir);
+        int dirExists = 0==stat(MMC_STRINGDATA(dir), &stat_buf);
+        sprintf(buf, "%s/%s", MMC_STRINGDATA(dir), resourcesDir);
         if (!dirExists || 0==stat(buf, &stat_buf)) {
           dir = buf;
         } else {
-          omc_assert_warning(info, PATH_NOT_IN_FMU_RESOURCES, dir);
+          omc_assert_warning(info, PATH_NOT_IN_FMU_RESOURCES, MMC_STRINGDATA(dir));
```

Comment:
@sjoelund Do you think this looks right? Specifically if the sprintf call was working before it might change behavior now.

Reply:
If dir is indeed a void* and not char*, that strlen would be wrong. I'm a little uncertain what types the sessions are and if they are different in static and dynamic FMU

Reply:
It seems like it is intended to be `modelica_string `(i.e., `void`*). The `strlen `should also be `MMC_STRLEN(dir)` I guess.

Reply:
I was a little confused because  (I guess) it was apparently working as intended. Is the function not used?

Reply:
It's used very rarely. Only in FMUs (since you cannot know the location the FMU resources directory is going to be unzipped to until runtime).

Reply:
Alright. Then I will fix the `strlen` as well and push this in.